tests/test_helper.hpp

```diff
@@ -74,8 +89,13 @@ bool almost_equal(T1<T> x, T2<T> y, int ulp) {
                  std::is_same<T2<T>, sycl::ext::cplx::complex<T>>::value),
                 "almost_equal should be used to compare complex number");
   // Somebody should be smart enough to refactor to use `enable_if`...
-  return almost_equal(x.real(), y.real(), ulp) &&
-         almost_equal(x.imag(), y.imag(), ulp);
+  auto diff =
+      complex_magnitude(std::complex(x.real() - y.real(), x.imag() - y.imag()));
+  return diff <= std::numeric_limits<T>::epsilon() *
+                     std::abs(std::complex(x.real() + y.real(),
+                                           x.imag() + y.imag())) *
+                     ulp ||
+         diff < std::numeric_limits<T>::min() || is_nan_or_inf(x, y);
```

Comment:
I think this is equivalent to `std::abs(x+y)`

Reply:
Yes done, same as above.